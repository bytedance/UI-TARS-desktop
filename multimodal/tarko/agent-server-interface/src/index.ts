/*
 * Copyright (c) 2025 Bytedance, Inc. and its affiliates.
 * SPDX-License-Identifier: Apache-2.0
 */

export * from '@tarko/agent-interface';
export * from './server';
export * from './config';
export * from './cli';
export * from './implementation';
export * from './constants';
