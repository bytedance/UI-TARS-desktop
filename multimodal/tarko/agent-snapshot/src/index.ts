/*
 * Copyright (c) 2025 Bytedance, Inc. and its affiliates.
 * SPDX-License-Identifier: Apache-2.0
 */

// Main API
export { AgentSnapshot } from './agent-snapshot';
export { AgentSnapshotRunner } from './agent-snapshot-runner';
export { SnapshotManager } from './snapshot-manager';

// Types
export type {
  AgentSnapshotOptions,
  VerificationOptions,
  SnapshotGenerationResult,
  SnapshotTestResult,
  TestRunConfig,
  ToolCallData,
  SnapshotCaseConfig,
  SnapshotCase,
  SnapshotInfo,
  // Backward compatibility
  SnapshotRunResult,
  CaseConfig,
} from './types';

// Utilities
export { AgentSnapshotNormalizer } from './utils/snapshot-normalizer';
export type { AgentNormalizerConfig, ComparisonResult } from './utils/snapshot-normalizer';
export { logger } from './utils/logger';

// Advanced/Internal API (for extension)
export { AgentHookBase } from './agent-hook-base';
export type { HookOptions } from './agent-hook-base';
export { AgentGenerateSnapshotHook } from './agent-generate-snapshot-hook';
export { AgentReplaySnapshotHook } from './agent-replay-snapshot-hook';
