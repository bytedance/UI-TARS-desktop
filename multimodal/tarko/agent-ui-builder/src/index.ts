/*
 * Copyright (c) 2025 Bytedance, Inc. and its affiliates.
 * SPDX-License-Identifier: Apache-2.0
 */

export * from './types';
export * from './builder';
export * from './static-path';

// Import for convenience exports
import { AgentUIBuilder } from './builder';
import type { AgentUIBuilderInputOptions, AgentUIBuilderResult, PostProcessor } from './types';

// Convenience exports for backward compatibility
export const buildHTMLInMemory = (input: AgentUIBuilderInputOptions): Promise<AgentUIBuilderResult> =>
  AgentUIBuilder.buildHTML(input, { destination: 'memory' });

export const buildHTMLToFile = (
  input: AgentUIBuilderInputOptions,
  filePath: string,
  overwrite = false,
): Promise<AgentUIBuilderResult> =>
  AgentUIBuilder.buildHTML(input, {
    destination: 'file',
    fileSystem: { filePath, overwrite },
  });

export const buildHTMLWithProcessor = (
  input: AgentUIBuilderInputOptions,
  postProcessor: PostProcessor,
): Promise<AgentUIBuilderResult> =>
  AgentUIBuilder.buildHTML(input, {
    destination: 'custom',
    postProcessor,
  });

export const generateDefaultFilePath = AgentUIBuilder.generateDefaultFilePath;
export const createShareProviderProcessor = AgentUIBuilder.createShareProviderProcessor;
