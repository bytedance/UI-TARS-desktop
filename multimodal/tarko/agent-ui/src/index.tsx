import React from 'react';

export const Home = () => <div>Home Component</div>;
export const App = (props: { navbar?: React.ReactNode; sidebar?: React.ReactNode; onBeforeInit?: () => void }) => (
  <div>
    {props.navbar}
    {props.sidebar}
    <div>App Content</div>
  </div>
);
export const Navbar = (props: { items?: React.ReactNode }) => <div>Navbar {props.items}</div>;
export const Sidebar = (props: { items?: React.ReactNode }) => <div>Sidebar {props.items}</div>;