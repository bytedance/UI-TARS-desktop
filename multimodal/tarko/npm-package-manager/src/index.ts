/*
 * Copyright (c) 2025 Bytedance, Inc. and its affiliates.
 * SPDX-License-Identifier: Apache-2.0
 */

export { TarkoNPMPackageManager } from './TarkoNPMPackageManager';
export type {
  PackageInfo,
  InstallOptions,
  AgentInputType,
} from './TarkoNPMPackageManager';
export { isNPMPackage, resolveAgentFromNPMInput } from './utils';
