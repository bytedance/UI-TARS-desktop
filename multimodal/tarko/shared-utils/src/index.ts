export * from './logger';
export * from './deepMerge';
export * from './env';
export * from './filter';
