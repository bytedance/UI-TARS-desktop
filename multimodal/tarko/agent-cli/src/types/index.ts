/*
 * Copyright (c) 2025 Bytedance, Inc. and its affiliates.
 * SPDX-License-Identifier: Apache-2.0
 */

import { AgentAppConfig, AgentConstructor } from '@tarko/agent-server-interface';
import { AgentServerExtraOptions } from '@tarko/agent-server';
import { Command } from 'cac';

export type { AgentServerExtraOptions };

/**
 * Command handler interface
 * Defines the structure for command handlers that process CLI commands
 */
export interface CommandHandler {
  /**
   * Execute the command with given options
   *
   * @param options Command options from CLI
   * @returns Promise that resolves when command completes
   */
  execute(options: Record<string, any>): Promise<void>;
}

/**
 * Server options for starting the web UI
 */
export interface WebUIOptions {
  /**
   * Complete application configuration
   */
  appConfig: AgentAppConfig;

  /**
   * Enable debug mode
   */
  isDebug?: boolean;

  /**
   * Agent constructor
   */
  agentConstructor: AgentConstructor;

  /**
   * Agent name for display
   */
  agentName: string;

  /**
   * Path to static files for web UI
   */
  staticPath?: string;

  /**
   * Extra options for the agent server
   */
  extraOptions?: AgentServerExtraOptions;
}

/**
 * Request options for sending direct requests to LLM providers
 */
export interface RequestOptions {
  /**
   * LLM provider name
   */
  provider: string;

  /**
   * Model name
   */
  model: string;

  /**
   * Path to request body JSON file or JSON string
   */
  body: string;

  /**
   * Custom API key
   */
  apiKey?: string;

  /**
   * Custom base URL
   */
  baseURL?: string;

  /**
   * Enable streaming mode
   */
  stream?: boolean;

  /**
   * Enable reasoning mode
   */
  thinking?: boolean;

  /**
   * Output format
   * - 'raw': Raw JSON output
   * - 'semantic': Human-readable formatted output
   * @default 'raw'
   */
  format?: 'raw' | 'semantic';
}

/**
 * Agent resolution result
 */
export interface AgentResolutionResult {
  /**
   * Agent constructor function
   */
  agentConstructor: AgentConstructor;

  /**
   * Agent name for logging
   */
  agentName: string;
}

/**
 * Bootstrap CLI options
 */
export interface TarkoAgentCLIOptions {
  /**
   * Version string
   */
  version: string;

  /**
   * Build time
   */
  buildTime: number;

  /**
   * Git hash
   */
  gitHash: string;

  /**
   * Binary name
   */
  binName?: string;

  /**
   * Custom agent constructor resolver
   */
  agentResolver?: AgentConstructorResolver;

  /**
   * Additional custom commands
   */
  customCommands?: CustomCommand[];

  /**
   * Remote configuration URL
   */
  remoteConfig?: string;
}

/**
 * Agent constructor resolver function
 */
export type AgentConstructorResolver = (
  agentParam: string,
) => Promise<AgentResolutionResult> | AgentResolutionResult;

/**
 * Custom command definition
 */
export interface CustomCommand {
  /**
   * Command name
   */
  name: string;

  /**
   * Command description
   */
  description: string;

  /**
   * Command handler
   */
  handler: CommandHandler;

  /**
   * Options configurator function
   */
  optionsConfigurator?: (command: Command) => Command;
}

/**
 * Run options for silent execution
 */
export interface RunOptions {
  appConfig: AgentAppConfig;
  input: string;
  agentConstructor: AgentConstructor;
  agentName: string;
  format?: 'json' | 'text';
  includeLogs?: boolean;
  isDebug?: boolean;
  agentServerExtraOptions: AgentServerExtraOptions;
}
