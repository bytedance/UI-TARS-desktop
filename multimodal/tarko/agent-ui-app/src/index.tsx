import React from 'react';
import { Home, App, Navbar, Sidebar } from '@tarko/agent-ui';

const Layout = () => (
  <App
    onBeforeInit={() => {}}
    navbar={<Navbar items={<><>} />}
    sidebar={<Sidebar items={<><>} />}
  />
);

export const CustomHome = () => <Home />;