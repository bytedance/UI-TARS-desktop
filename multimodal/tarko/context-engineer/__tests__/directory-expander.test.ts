import { it, expect } from 'vitest';
import { DirectoryExpander } from '../src';
import path from 'path';

const AgentServer = path.join(__dirname, '../../agent-server/src');

it('DirectoryExpander', async () => {
  const directoryExpander = new DirectoryExpander();
  const result = await directoryExpander.expandDirectories(['src'], AgentServer);
  expect(result).toMatchSnapshot();
});
