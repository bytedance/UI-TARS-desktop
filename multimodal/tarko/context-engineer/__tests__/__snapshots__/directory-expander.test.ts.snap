// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`DirectoryExpander 1`] = `
{
  "expandedContent": "",
  "files": [],
  "processedDirectories": [],
  "stats": {
    "errorCount": 0,
    "totalFiles": 0,
    "totalSize": 0,
  },
}
`;
